daemon/cmd/datapath.go

```diff
@@ -367,3 +372,35 @@ func setupRouteToVtepCidr(logger *slog.Logger) error {
 
 	return nil
 }
+
+type k8sHostFirewallBypass struct{}
+
+func newK8sHostFirewallBypass(config *option.DaemonConfig) k8sClient.ConfigureK8sClientsetDialer {
+	if config.EnableK8sHostFirewallBypass {
+		return &k8sHostFirewallBypass{}
+	} else {
+		return nil
+	}
+}
+
+// Sets SO_MARK so that connections to kube-apiserver bypass host firewall and DNS proxy
+func (*k8sHostFirewallBypass) ConfigureK8sClientsetDialer(dialer *net.Dialer) {
+	dialer.Control = setProxyEgressMark
+	dialer.Resolver = &net.Resolver{
+		PreferGo: true,
```

Comment:
question: was this the case prior to this change?  If not, what are the implications of having this be true?

Reply:
It was. [Go documentation](https://pkg.go.dev/net#hdr-Name_Resolution) says that on Unix, the pure go resolver is preferred. I put this in anyway to express intent (setting the resolver's dialer makes no sense unless the pure go resolver is used) and also in case somebody runs the daemon with `GODEBUG=netdns=cgo` or similar.

Reply:
Ok, thanks for the clarification.